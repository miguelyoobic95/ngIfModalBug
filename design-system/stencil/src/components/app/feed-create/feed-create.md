---
name: Feed Create
category: Core Components
---

```feed-create.html
    <yoo-feed-create></yoo-feed-create>
```

## Attributes

|Name|Type|Default|Description|
|---|---|---|---|
|`text`|string|   |the text inside the label|

## CSS

|Type|Name|Description|
|---|---|---|


## Events
|Attr|Description|
|---|---|---|
|`yooXXXX`|small label|