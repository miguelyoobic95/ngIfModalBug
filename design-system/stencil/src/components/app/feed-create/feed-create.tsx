import { Component, Element, Prop } from '@stencil/core';

@Component({
    tag: 'yoo-feed-create',
    styleUrl: 'feed-create.scss',
    scoped: true
})
export class YooFeedCreateComponent {

     @Element() host: HTMLStencilElement;

    @Prop() photoItems: Array<any>;

    render(): JSX.Element {
        return (
            <div>{
                (this.photoItems || []).map(photo => <img class="thumbnail" src={photo.thumbnailURL} />)
            }</div>
        );
    }
}
