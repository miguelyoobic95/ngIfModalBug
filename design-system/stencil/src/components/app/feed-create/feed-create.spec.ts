import { TestWindow } from '@stencil/core/testing';
import { YooFeedCreateComponent } from './feed-create';

describe('FeedCreateComponent', () => {
    it('should build', () => {
        expect(new YooFeedCreateComponent()).toBeTruthy();
    });

    describe('Rendering', () => {
        let element, window;
        beforeEach( async () => {
            window = new TestWindow();
            element = await window.load({
                components: [YooFeedCreateComponent],
                html: '<yoo-feed-create></yoo-feed-create>'
            });
        });

        it ('Should render', async () => {
            expect(element).toMatchSnapshot();
        });
    });

    describe('Events', () => {

        let emitEvent, window;
        beforeEach( async () => {
            window = new TestWindow();
            emitEvent = jest.fn();
        });
    });
});