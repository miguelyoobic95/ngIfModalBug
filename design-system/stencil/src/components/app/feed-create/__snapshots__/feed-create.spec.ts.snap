// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FeedCreateComponent Rendering Should render 1`] = `
<yoo-feed-create
  class="hydrated"
  data-yoo-feed-create-host=""
>
  
  <div
    data-yoo-feed-create=""
  />
</yoo-feed-create>
`;
